presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/RewriteAggregationIfToFilter.java

```diff
@@ -199,4 +225,23 @@ private boolean shouldRewriteAggregation(Aggregation aggregation, ProjectNode so
         // Only rewrite the aggregation if the else branch is not present or the else result is NULL.
         return expression.getForm() == IF && Expressions.isNull(expression.getArguments().get(2));
     }
+
+    private boolean canUnwrapIf(SpecialFormExpression ifExpression)
+    {
+        boolean[] result = {true};
+        ifExpression.getArguments().get(1).accept(new DefaultRowExpressionTraversalVisitor<boolean[]>()
+        {
+            @Override
+            public Void visitCall(CallExpression call, boolean[] result)
+            {
+                Optional<OperatorType> operatorType = functionAndTypeManager.getFunctionMetadata(call.getFunctionHandle()).getOperatorType();
+                if (operatorType.isPresent() && (operatorType.get() == OperatorType.DIVIDE || operatorType.get() == OperatorType.SUBSCRIPT)) {
```

Comment:
Shall we add a comment. Plus, do we see other cases that we should disable the rewrite?

Reply:
Added a comment. There may be other cases. But not that I am aware of now. 
Deltoid doesn't have any of these cases. So it should be safe to enable it for Deltoid clusters.